ethcore/src/engines/clique/mod.rs

```diff
@@ -181,33 +178,38 @@ pub struct Clique {
 	pub block_state_by_hash: RwLock<LruCache<H256, CliqueBlockState>>,
 	pub proposals: RwLock<HashMap<Address, VoteType>>,
 	pub signer: RwLock<Option<Box<EngineSigner>>>,
-	pub step_service: Option<StepService>,
 }
 
 impl Clique {
 	/// Initialize Clique engine from empty state.
 	pub fn new(params: CliqueParams, machine: EthereumMachine) -> Result<Arc<Self>, Error> {
-		let mut engine = Clique {
+		/// Pause before starting to step Clique
+		const INITIAL_DELAY: Duration = Duration::from_secs(5);
+		/// Step Clique at most every 2 seconds
+		const SEALING_FREQ: Duration = Duration::from_secs(2);
```

Comment:
That seems way too big, if you configure step to be `5seconds` you will end up being one second (20%) late in a worst case.

Reply:
@niklasad1 Did the stepping frequency come from the spec? I tried using 200ms here and got plenty of `TRACE miner  requires_reseal: reseal too early` in the logs.

Reply:
@dvdplm that's because of other setting from `miner` (`--min-reseal-period`) - it prevents constructing new blocks too often (usualy it's due to new transactions coming into the pool). But it seems pretty weird that apparently `step()` actually constructs new block every time? Then shortening the time will be pretty grim from performance perspective especially in networks where blocks are at it's limits.

Reply:
@dvdplm It is not part of the spec AFAIK, as Tomek said I think it is used for sealing timing.